src/utils/breaches.ts

```diff
@@ -0,0 +1,236 @@
+ * //
+ *
+ * @param foundBreaches
+ */
+function addRecencyIndex(foundBreaches: HibpLikeDbBreach[]) {
+  // /**
+  //  * @type {any[]}
+  //  */
+  const annotatedBreaches: HibpLikeDbBreach[] = [];
+  // slice() the array to make a copy so before reversing so we don't
+  // reverse foundBreaches in-place
+  const oldestToNewestFoundBreaches = foundBreaches.slice().reverse();
+  oldestToNewestFoundBreaches.forEach((annotatingBreach, index) => {
+    const foundBreach = foundBreaches.find(
+      (foundBreach) => foundBreach.Name === annotatingBreach.Name,
+    );
+    annotatedBreaches.push(Object.assign({ recencyIndex: index }, foundBreach));
+  });
+  return annotatedBreaches.reverse();
+}
+
+/**
+ * TODO: deprecate with MNTOR-2021
+ *
+ * @param {{ user: any; email: any; recordId: any; recordVerified: any; allBreaches: import('./hibp').HibpLikeDbBreach[]; }} options
+ * @returns {Promise<BundledVerifiedEmails>}
+ */
+
+type options = {
+  user: userType;
+  verifiedEmail: BundledVerifiedEmails;
+  recordId: number;
+  recordVerified: boolean;
+  allBreaches: HibpLikeDbBreach[] | SubscriberBreach[];
+};
+async function bundleVerifiedEmails(
+  options: options,
+): Promise<BundledVerifiedEmails> {
+  const { user, verifiedEmail, recordId, recordVerified, allBreaches } =
+    options;
+  const lowerCaseEmailSha = getSha1(verifiedEmail.email.toLowerCase());
+
+  // find all breaches relevant to the current email
+  const foundBreaches = await getBreachesForEmail(
+    lowerCaseEmailSha,
+    allBreaches,
+    true,
+    false,
+  );
+
+  // TODO: remove after migration MNTOR-978
+  // adding index to breaches based on recency
+  const foundBreachesWithRecency = addRecencyIndex(foundBreaches);
+
+  if (!user) {
+    const errMsg = "breachResolutionV2: subscriber cannot be undefined";
+    console.error(errMsg);
+    captureMessage(errMsg);
+
+    // @ts-ignore: function will be deprecated
+    return { verifiedEmails, unverifiedEmails };
+  }
+
+  // get v2 "breach_resolution" object
+  const breachResolutionV2 = user.breach_resolution
+    ? user.breach_resolution[verifiedEmail.email]
+      ? user.breach_resolution[verifiedEmail.email]
+      : {}
+    : [];
+
+  const useBreachId = user.breach_resolution?.useBreachId;
+
+  for (const breach of foundBreachesWithRecency) {
+    // if breach resolution json has `useBreachId` boolean, that means the migration has taken place
+    // we will use breach id as the key. Otherwise, we fallback to using recency index for backwards compatibility
+    if (useBreachId) {
+      breach.IsResolved = breachResolutionV2[breach.Id]?.isResolved;
+      breach.ResolutionsChecked =
+        breachResolutionV2[breach.Id]?.resolutionsChecked || [];
+    } else {
+      // TODO: remove after MNTOR-978
+      breach.IsResolved = breachResolutionV2[breach.recencyIndex]?.isResolved;
+      breach.ResolutionsChecked =
+        breachResolutionV2[breach.recencyIndex]?.resolutionsChecked || [];
+    }
+
```

Comment:
Yeah, I think instead of trying to modify the `foundBreachesWithRecency` array (of type `HibpLikeDbBreach[]`) in place to try to make it look like `SubscriberBreach[]`, it's probably best to just create a new array of `SubscriberBreach`es.